x/evm/genesis.go

```diff
@@ -77,19 +77,27 @@ func InitGenesis(
 func ExportGenesis(ctx sdk.Context, k *keeper.Keeper, ak types.AccountKeeper) *types.GenesisState {
 	var ethGenAccounts []types.GenesisAccount
 	ak.IterateAccounts(ctx, func(account authtypes.AccountI) bool {
-		ethAccount, ok := account.(evmostypes.EthAccountI)
+		acc, ok := account.(*authtypes.BaseAccount)
 		if !ok {
-			// ignore non EthAccounts
 			return false
 		}
 
-		addr := ethAccount.EthAddress()
+		address, err := utils.Bech32ToHexAddr(acc.Address)
+		if err != nil {
+			return false
+		}
+
+		codeHash := k.GetCodeHash(ctx, address)
+		if !types.BytesAreEmptyCodeHash(codeHash.Bytes()) {
```

Comment:
this should return if the account does not have code hash, which means it is empty code hash, right?
```suggestion
		if types.BytesAreEmptyCodeHash(codeHash.Bytes()) {
```

Reply:
test is now added and logic adjusted 👍